cranelift-filetests/src/test_unwind.rs

```diff
@@ -0,0 +1,233 @@
+
+        let mut text = String::new();
+        if mem.len() == 0 {
+            writeln!(text, "No unwind information.").unwrap();
+        } else {
+            print_unwind_info(&mut text, &mem);
+        }
+
+        run_filecheck(&text, context)
+    }
+}
+
+fn print_unwind_info(text: &mut String, mem: &[u8]) {
+    assert!(mem.len() >= core::mem::size_of::<UnwindInfo>());
+    let info = unsafe { UnwindInfo::from_ptr(mem.as_ptr()) };
+
+    let node_count = info.unwind_code_count();
+
+    // Assert correct alignment and padding of the unwind information
+    assert!(mem.len() % 4 == 0);
+    assert_eq!(
+        mem.len(),
+        core::mem::size_of::<UnwindInfo>()
+            + ((node_count as usize) * core::mem::size_of::<UnwindCode>())
+            + if (node_count & 1) == 1 {
+                core::mem::size_of::<UnwindCode>()
+            } else {
+                0
+            }
+    );
+
+    writeln!(
+        text,
+        "Unwind info: version = {}, flags = {}, prologue size: {}, unwind code count = {}, frame register = {}, frame register offset = {}",
+        info.version(),
+        info.flags(),
+        info.prologue_size(),
+        node_count,
+        info.frame_register(),
+        info.frame_register_offset()
+    ).unwrap();
+
+    let mut i = 0;
+    while i < node_count {
+        let offset = (core::mem::size_of::<UnwindInfo>()
+            + ((i as usize) * core::mem::size_of::<UnwindCode>())) as isize;
+        let code = unsafe { UnwindCode::from_ptr(mem.as_ptr().offset(offset)) };
+
+        write!(
+            text,
+            "Unwind code: offset = {}, operation = {:#?}, info = {}",
+            code.offset(),
+            code.operation(),
+            code.info(),
+        )
+        .unwrap();
+
+        match code.additional_nodes() {
+            0 => {
+                writeln!(text).unwrap();
+                i += 1;
+            }
+            1 => {
+                let offset = offset + core::mem::size_of::<UnwindCode>() as isize;
+                writeln!(text, ", value = {}", unsafe {
+                    core::ptr::read_unaligned(mem.as_ptr().offset(offset) as *const u16)
+                })
+                .unwrap();
+                i += 2;
+            }
+            2 => {
+                let offset = offset + core::mem::size_of::<UnwindCode>() as isize;
+                writeln!(text, ", value = {}", unsafe {
+                    core::ptr::read_unaligned(mem.as_ptr().offset(offset) as *const u32)
+                })
+                .unwrap();
+                i += 3;
+            }
+            _ => panic!("unexpected number of additional nodes"),
+        };
+    }
+}
+
+#[repr(C)]
+struct UnwindInfo {
+    version_and_flags: u8,
+    prologue_size: u8,
+    unwind_code_count: u8,
+    frame_register: u8,
+}
+
+impl UnwindInfo {
+    unsafe fn from_ptr(ptr: *const u8) -> Self {
+        core::ptr::read_unaligned(ptr as *const Self)
```

Comment:
Please take a slice and read the individual fields instead.

Reply:
Agreed, that would be better.  I'll fix.